.github/workflows/tests.yaml

```diff
@@ -0,0 +1,79 @@
+---
+name: Testing
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+env:
+  DEFAULT_PYTHON: "3.11"
+
+jobs:
+  pytest:
+    name: Python ${{ matrix.python }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ["3.11", "3.12"]
+    steps:
+      - name: ⤵️ Check out code from GitHub
+        uses: actions/checkout@v4.1.4
+      - name: 🏗 Set up Poetry
+        run: pipx install poetry
+      - name: 🏗 Set up Python ${{ matrix.python }}
+        id: python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: ${{ matrix.python }}
+          cache: "poetry"
+      - name: 🏗 Install workflow dependencies
+        run: |
+          poetry config virtualenvs.create true
+          poetry config virtualenvs.in-project true
+      - name: 🏗 Install dependencies
+        run: poetry install --no-interaction
+      - name: Lint with flake8
+        run: |
+          poetry run flake8 src/forecast_solar
+      - name: 🚀 Run pytest
+        run: poetry run pytest -v --cov src tests
+      - name: ⬆️ Upload coverage artifact
+        uses: actions/upload-artifact@v4.3.3
+        with:
+          name: coverage-${{ matrix.python }}
+          path: .coverage
+
+  coverage:
+    runs-on: ubuntu-latest
+    needs: pytest
+    steps:
+      - name: ⤵️ Check out code from GitHub
+        uses: actions/checkout@v4.1.4
+        with:
+          fetch-depth: 0
+      - name: ⬇️ Download coverage data
+        uses: actions/download-artifact@v4.1.7
+      - name: 🏗 Set up Poetry
+        run: pipx install poetry
+      - name: 🏗 Set up Python ${{ env.DEFAULT_PYTHON }}
+        id: python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+          cache: 'poetry'
+      - name: 🏗 Install workflow dependencies
+        run: |
+          poetry config virtualenvs.create true
+          poetry config virtualenvs.in-project true
+      - name: 🏗 Install dependencies
+        run: poetry install --no-interaction
+      - name: 🚀 Process coverage results
+        run: |
+          poetry run coverage combine coverage*/.coverage*
```

Comment:
Is this really needed? It's not possible to just upload all the files?

Reply:
I think it already makes 1 coverage file by default, so I could skip combining it indeed

Reply:
It ensures that both coverage artifacts (python 3.11 and 3.12) from the first job are combined into one file and then uploaded as xml to codecov.

Reply:
We can remove this.